app/models/student.rb

```diff
@@ -141,45 +138,33 @@ def create_group_for_working_alone_student(aid)
         end
       end
 
-      @grouping.group = @group
-      begin
-        unless @grouping.save
-          m_logger = MarkusLogger.instance
-          m_logger.log("Could not create a grouping for Student '#{user_name}'"\
+      empty_grouping = Grouping.find_by(assignment_id: aid, group_id: @group.id)
+      unless empty_grouping.nil?
+        # if this is found, then the student must have an (empty) existing grouping that he is not a member of
+        # this can happen if an instructor removes the student from its grouping (see issue 627)
+        # we must delete this grouping first
+        empty_grouping.destroy
+      end
+      @grouping = Grouping.new(assignment_id: aid, group_id: @group.id)
+      unless @grouping.save
+        m_logger = MarkusLogger.instance
+        m_logger.log("Could not create a grouping for Student '#{user_name}'"\
           ". The grouping was:  #{@grouping.inspect} - errors: "\
           "#{@grouping.errors.inspect}", MarkusLogger::ERROR)
-          raise 'Sorry!  For some reason, your grouping could not be created. '\
+        raise 'Sorry!  For some reason, your grouping could not be created. '\
           ' Please wait a few seconds, and hit refresh to try again.  If you'\
           ' come back to this page, you should inform the course instructor.'
-        end
-      # This exception will only be thrown when we try to save to a grouping that already exists
-      rescue ActiveRecord::RecordNotUnique
-        # transaction has failed, so quit it
-        return false
-      end
-
-      # We give students the tokens for the test framework
-      if @assignment.enable_test
```

Comment:
Here I'm just checking that this happens somewhere else? Or is unnecessary?

Reply:
It's unnecesary, tokens are given at test request time if needed.
(but thinking about it, let me double check the interface instead, whether it hides the button without tokens)

Reply:
All good!